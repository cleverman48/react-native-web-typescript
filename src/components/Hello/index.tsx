import * as React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

export interface IProps {
  name: string;
  enthusiasmLevel?: number;
}

interface IState {
  enthusiasmLevel: number;
}

export default class Hello extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    if ((props.enthusiasmLevel || 0) <= 0) {
      throw new Error("You could be a little more enthusiastic. :D");
    }

    this.state = {
      enthusiasmLevel: props.enthusiasmLevel || 1
    };
  }

  public onIncrement = () =>
    this.setState({ enthusiasmLevel: this.state.enthusiasmLevel + 1 });
  public onDecrement = () =>
    this.setState({
      enthusiasmLevel:
        this.state.enthusiasmLevel - 1 > 0 ? this.state.enthusiasmLevel - 1 : 0
    });
  public getExclamationMarks = (numChars: number) =>
    Array(numChars + 1).join("!");

  public render() {
    return (
      <View style={styles.root}>
        <Text style={styles.greeting}>
          Hello{" "}
          {this.props.name +
            this.getExclamationMarks(this.state.enthusiasmLevel)}
        </Text>

        <View style={styles.buttons}>
          <View style={styles.button}>
            <Button
              title="-"
              onPress={this.onDecrement}
              accessibilityLabel="decrement"
            />
          </View>

          <View style={styles.button}>
            <Button
              title="+"
              onPress={this.onIncrement}
              accessibilityLabel="increment"
            />
          </View>
        </View>
      </View>
    );
  }
}

// styles
const styles = StyleSheet.create({
  root: {
    alignItems: "center",
    alignSelf: "center"
  },
  buttons: {
    flexDirection: "row"
  },
  button: {
    flex: 1,
    paddingVertical: 0
  },
  greeting: {
    color: "#999",
    fontWeight: "bold"
  }
});
