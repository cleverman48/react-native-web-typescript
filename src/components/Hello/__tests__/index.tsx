import * as React from "react";
import { create } from "react-test-renderer";
import Hello from "../";

describe("Hello", () => {
  it("renders correctly with defaults", () => {
    const button = create(<Hello name="World" enthusiasmLevel={1} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("increments", () => {
    const button = create(
      <Hello name="World" enthusiasmLevel={1} />
    ).getInstance();
    button.onIncrement();
    button.onIncrement();
    expect(button.state.enthusiasmLevel).toBe(3);
  });

  it("decrements", () => {
    const button = create(
      <Hello name="World" enthusiasmLevel={4} />
    ).getInstance();
    button.onDecrement();
    button.onDecrement();
    expect(button.state.enthusiasmLevel).toBe(2);
  });

  it("decrements at zero", () => {
    const button = create(
      <Hello name="World" enthusiasmLevel={0} />
    ).getInstance();
    button.onDecrement();
    expect(button.state.enthusiasmLevel).toBe(0);
  });
});
