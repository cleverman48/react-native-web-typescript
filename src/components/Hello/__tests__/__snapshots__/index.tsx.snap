// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#999",
        "fontWeight": "bold",
      }
    }
  >
    Hello
     
    World!
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 0,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel="decrement"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 0,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel="increment"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
